nanoFramework.Hardware.Esp32.Rmt/ReceiverChannel.cs

```diff
@@ -13,79 +13,79 @@ namespace nanoFramework.Hardware.Esp32.Rmt
     /// </summary>
     public class ReceiverChannel : RmtChannel, IDisposable
     {
-        #region Fields
-
         [System.Diagnostics.DebuggerBrowsable(System.Diagnostics.DebuggerBrowsableState.Never)]
-        private TimeSpan _receiveTimeout = new TimeSpan(0, 0, 1); // 1 sec
+        private readonly ReceiverChannelSettings _receiverChannelSettings;
 
-        #endregion Fields
+        /// <inheritdoc/>
+        public override ChannelMode Mode => ChannelMode.Receive;
 
         /// <summary>
-        /// The receive time-out used when calling GetAllItems. Default 1 second.
+        /// Gets or sets the idle threshold after which the receiver will go into idle mode 
+        /// and <see cref="RmtCommand"/>s are copied into the ring buffer and availble to your code.
         /// </summary>
-        public TimeSpan ReceiveTimeout
+        public ushort IdleThreshold
```

Comment:
I was looking at this when reviewing the native end. 
Having it as a `TimeSpan` looks more C# and makes it much clearer than using an `ushort`.
Would you agree @Ellerbach ?

Reply:
Not entirely sure about this suggestion. The Idle Receive Threshold in IDF is the clock cycles after applying a divider. If the divider value is 80 then yes, we can easily map the timespan value to ticks since each clock cycle is one microsecond. But, if the dev puts another divider then we have to calculate it before passing the value to IDF.

Happy to hear @Ellerbach feedback on this one too.

Reply:
My concern, again, it's usability. Let's put this timespan idea aside for a moment. 
The developer has to know that we're clocking this at 80MHz and that this is a "multiplier" of the period corresponding to this. 
All of it to "simply" have this as the setting for which "The receiver will ignore pulses longer than this setting." as the IDF documentation states.

Maybe we could shamelessly copy paste the description from IDF docs and add a remark making it clear what is this for. Something like (please improve it 😁  ) : RMT clock is set to 80MHz, therefore this will make the receiver ignore pulses wider than 1/80M x n or similar. 

Also as a general guidance we could have a note on the class (and maybe in the readme too). For detailed explainations please check Espressif official documentation here https://docs.espressif.com/projects/esp-idf/en/v4.4/esp32/api-reference/peripherals/rmt.html

Reply:
To be honest, I am on the side of keeping things close to IDF and improving the docs for that property. This way users of this library can refer to either the nf docs or IDF docs and would be able to easily navigate how to use it. I'll update the comments but let me know what you think :)